src/Compilers/CSharp/Portable/BoundTree/BoundTreeRewriter.cs

```diff
@@ -131,13 +135,13 @@ public sealed override BoundNode VisitBinaryOperator(BoundBinaryOperator node)
                 binary = (BoundBinaryOperator)child;
             }
 
-            var left = (BoundExpression)this.Visit(child);
+            var left = (BoundExpression?)this.Visit(child)!;
 
             do
             {
                 binary = stack.Pop();
-                var right = (BoundExpression)this.Visit(binary.Right);
-                var type = this.VisitType(binary.Type);
+                var right = (BoundExpression?)this.Visit(binary.Right)!;
+                var type = this.VisitType(binary.Type)!;
```

Comment:
>this.VisitType(binary.Type)!; [](start = 27, length = 29)

I didn't understand why that suppression is necessary or the `?` in cast above.

Reply:
They aren't, now that I've annotated `VisitType`.  Will revert.

---
In reply to: [386578093](https://github.com/dotnet/roslyn/pull/42068#discussion_r386578093) [](ancestors = 386578093)